### ISSUE TITLE
False positive for “RET504: Unnecessary variable assignment before `return` statement”

### ISSUE THREAD

**Issue by quassy** [charliermarsh/ruff#1159]
Code similar to the following will return a false positive for RET504, as the variable assignment here is in fact not unnecessary or optional:
```python
def foobar() -> str:
    result = get_foobar_from_db()
    print(f"Got {len(result)} number of results from db.")
    return result
```

**Comment by charliermarsh** [comment]
Good catch.
